### PR TITLE
[INFRA-288] Fix Java parsing and generate tests

### PR DESCRIPTION
**Pull request by jayvdb**

### CODE REVIEW

File: generator/sbpg/targets/resources/MessageTest.java.j2 [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        org.junit.Assert.assertEquals(msg.(((prefix))), (((value|str_escape))) );
+((*- elif value is array_type *))
+((*- for ff in value *))((( compare_value( (((prefix))) + '[' + (((loop.index0|to_str))) + ']', (((ff))) ) )))((*- endfor *))
+((*- elif value is dict_type *))
+((*- for k in (((value|sorted))) *))((( compare_value( (((prefix))) + '.' + (((k))), (((value[k]))) ) )))((*- endfor *))
+((*- elif value is float_type *))((=
+    Note: the ("%.17e"|format(value)) filter is intended to preserve float
+    literal precision accross all value ranges. =))
+        org.junit.Assert.assertEquals(msg.(((prefix))), ((("%.17e"|format(value)))), DELTA);
+((*- elif value in [2159176030, 3411995557, 7909447587, 8409447265, 8907446923, 9406446591, 9906446269] *))
```
Comment (jayvdb) [inline generator/sbpg/targets/resources/MessageTest.java.j2:39]: These seven are somehow different, possibly a bug in the parsing.

File: generator/sbpg/targets/test_rust.py [hunk 2]
```diff
@@ -20,19 +20,13 @@
 
 TEST_TEMPLATE_NAME = "sbp_tests_template.rs"
 
-def b64_decode(field):
```
Comment (silverjam) [inline generator/sbpg/targets/test_rust.py:23]: Moved to common loc. or not used?

File: generator/sbpg/targets/resources/sbp_java.java.j2 [hunk 3]
```diff
@@ -52,7 +52,7 @@ public class (((m.identifier | classnameify))) extends SBPMessage {
     public (((m.identifier | classnameify))) () { super(TYPE); }
     public (((m.identifier | classnameify))) (SBPMessage msg) throws SBPBinaryException {
         super(msg);
-        assert msg.type != TYPE;
+        assert msg.type == TYPE;
```
Comment (silverjam) [inline generator/sbpg/targets/resources/sbp_java.java.j2:55]: 🤔 how did this ever work? Do asserts not fire in certain cases?

File: generator/sbpg/targets/resources/MessageTest.java.j2 (hunk 1 above)
Comment (silverjam) [reply generator/sbpg/targets/resources/MessageTest.java.j2:39]: These all look larger than what will fit in 32-bit unsigned int-- 🤷‍♀️

**Review by silverjam — COMMENTED** [review]
Looks promising so far, int parsing bug is a bit concerning, but we can possibly ship this and do a follow-up since the bug already exists

File: generator/sbpg/targets/resources/sbp_java.java.j2 (hunk 3 above)
Comment (jayvdb) [reply generator/sbpg/targets/resources/sbp_java.java.j2:55]: no ;-)

I assume the existing usage/examples doesnt go through this constructor, as it uses SBPHandler.

File: generator/sbpg/targets/test_rust.py (hunk 2 above)
Comment (jayvdb) [reply generator/sbpg/targets/test_rust.py:23]: Moved to .common

File: generator/sbpg/targets/resources/MessageTest.java.j2 (hunk 1 above)
Comment (jayvdb) [reply generator/sbpg/targets/resources/MessageTest.java.j2]: The first two were erroneously listed here; and the rest are all greater than unsigned int.
However the problem was endianness. It is only a few attributes where this crops up, and it silently failed, so not surprising nobody noticed this.

**Review by silverjam — APPROVED** [review]

**Pull request merged**